### PR TITLE
Add changelog for 0.7.0

### PR DESCRIPTION
**Pull request by psychocoderHPC**
Co-authored-by: Klaus Steiniger <k.steiniger@hzdr.de>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,354 @@
...
 =========
 
+0.7.0
+-----
+
+**Date:** 2023-10-XX
+
+C++17, lasers via incidentField, DispersivePulse laser profile, sub-stepping field solver, and [PICMI](https://picmi-standard.github.io/#) support
+
+With this release, the laser initialization switches to an incidentField approach, which allows easy initialization of arbitrary, Maxwell conform electromagnetic fields in a simulation. Old `laser.param` files won't work anymore. See the documentation!
```
Comment (PrometheusPi) [inline CHANGELOG.md]:
```suggestion
With this release, the laser initialization switches to an incidentField approach, which allows easier initialization of arbitrary, Maxwell conform, electromagnetic fields in a simulation. Old `laser.param` files won't work anymore. See the documentation!
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,354 @@
...
+    - Arbitrary order incident field solver #3860 #4038 #4050 #4082 #4105
+    - Add a runtime option to set particle density file #3972
+    - fix warnings and check for warnings in CI #3878
+    - Implement phase-coefficients in Gauss-Laguerre laser modes #3992
+    - Add a check and warning printed when no volume inside the Huygens surface #4049
+    - Add a warning for auto-adjusting domain to fit field absorber #4047
+    - Change way of controlling position of the Huygens surface #4056
+    - Check uniqueness of particle attributes #4078
+    - Improve reporting of wrong MPI/devices configuration #4122
+    - Check that averagized derived attributes are weighted #4154
```
Comment (PrometheusPi) [inline CHANGELOG.md]:
```suggestion
    - Check that averagely derived attributes are weighted #4154
```
This adverb is strange.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,354 @@
...
+- Specify BP5 default BufferChunkSize as 2GB #4127
+- Unify default openPMD file extension in plugins #4155
+- Add derived particle attributes for component of momentum and related density #4169
+- Add custom normalization option to PNG output #4165
+- Add a new functor to sample macroparticles in a cell #4111
+- Auto adjust particle exchange buffers for double precision #4236
+- Disallow binomial current interpolation together with current background #4252
+- Add free temperature functor #4256
+- Change user interface of free temperature functor to match free density functor #4273
+- [FLYonPIC] atomic physics for PIConGPU #4134
```
Comment (PrometheusPi) [inline CHANGELOG.md]:
```suggestion
- FLYonPIC atomic physics for PIConGPU #4134
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,354 @@
...
+    - Allow fractional indices in TWTSfast functors #4184
+    - Calculate and output laser phase velocity #4174
+    - Remove derived attribute/field MomentumComponent #4191
+    - HIP: use compute current strategy `strategy::CachedSupercellsScaled<2>` #4218
+    - Add a new functor to sample macroparticles in a cell #4111
+    - Add time-substepping versions of existing field solvers #3804
+    - Add time interpolation/extrapolation of J in substepping field solver  #4258
+    - Improve treating currents in PML #4293
+    - Add a new particle attribute for weighting damping factor  #4300
+    - [FLYonPIC] atomic physics for PIConGPU #4134
```
Comment (PrometheusPi) [inline CHANGELOG.md]:
```suggestion
    - FLYonPIC atomic physics for PIConGPU #4134
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,354 @@
...
+    - openPMD plugin: Set default backend depending on available options #4008
+    - openPMD plugin: throw error when neither defining period or toml #4028
+    - Clarify an inaccurate exception message in calorimeter plugin #4027
+    - Specify BP5 default BufferChunkSize as 2GB #4127
+    - Add custom normalization option to PNG output #4165
+    - OpenPMD plugin data range selection #4096
+    - Switch to .bp4 extension for ADIOS2 if available, print available extensions in help message #4234
+- tools:
+    - Crusher HIPCC Profile: Fix export of env variables #3976
+    - crusher hipcc profile #3973
```
Comment (PrometheusPi) [inline CHANGELOG.md]:
```suggestion
    - Crusher HIPCC profile #3973
```
same as above

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,6 +1,354 @@
...
+    - Fix copying of cuSTL device buffers #4029
+    - Fix function `toTimeSlice` #4040
+    - Fix implementation of pmacc::makeDeepCopy() and field background #4192
+    - Fix soft restart #4034
+    - Fix PMacc reduce #4328
+- plugins:
+    - Add missing communication in XrayScattering #3937
+    - Fix rad restart #3961
+    - Fix calculation of total current in the SumCurrents plugin for 2d case #4118
+    - Fix radiation trees. #4231
```
Comment (PrometheusPi) [inline CHANGELOG.md]:
```suggestion
    - Fix radiation trees #4231
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,6 +1,354 @@
...
+- Fix that PMacc depends on PIConGPU type definitions #3925
+- HIP: allow full architecture definition #3941
+- Make the span-based storeChunk API opt-in in various openPMD-based IO routines #3933
+- Fix openPMD outputs negative particle patch offset #4037
+- Fix CUDA container apt keys #4085
+- Add HIP and mallocMC + HIP to PIConGPU version output #4139
+- Do not forceinline recursive functions #4147
+- PNG plugin: fix invalid memory access in 2D #4225
+- Fix using uninitilized variable #4303
+- [breaking change] Fix openPMD particles `positionOffset` #4283
```
Comment (PrometheusPi) [inline CHANGELOG.md]:
```suggestion
- **breaking change** Fix openPMD particles `positionOffset` #4283
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,6 +1,354 @@
...
+        - Remove Cell2Particle #4289
+        - Remove getGlobalSuperCells #4325
+        - Move include collision.unitless to the default loader #4326
+        - Refactor lockstep programming #4321
+        - Switch to std::optional in compute field value #4345
+        - Use cupla/alpaka's min/max function #4350
+        - Remove flylite #4360
+        - Remove using CUSTL for particle creation #4363
+        - Remove Bremsstrahlung and Synchrotron radition extension #4395
+        - Remove CUstl from particle push #4398
```
Comment (PrometheusPi) [inline CHANGELOG.md]:
```suggestion
        - Remove cuSTL from particle push #4398
```
same as below

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,6 +1,354 @@
...
+        - Fix incident field calculation order #4451
+        - Improve command line argument validation #4447
+        - Change class RelayPoint to function #4464
+        - OnePositionImpl refactoring #4465
+        - Refactor particle pusher launcher #4471
+        - Refactor differentiation trait #4473
+        - Refactor GetChargeState and GetCharge #4474
+        - Remove usage of `boost::result_of` #4505
+    - PMacc:
+        - Clean/refactor CUSTL #3862
```
Comment (PrometheusPi) [inline CHANGELOG.md]: same as above 
```suggestion
        - Clean/refactor cuSTL #3862
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,6 +1,354 @@
...
+        - Refactor openPMD plugin #4054
+        - Improve comments and naming in PNG output #4164
+        - openPMD plugin: add IO file version #4287
+        - openPMD plugin: Flush data to disk within a step #4002
+        - openPMD plugin: JSON parameters for restarting #4329
+        - Use cuplas/alpakas mapped memory #4348
+        - Remove xray scattering #4352
+        - Remove plugin: PositionsParticle #4371
+        - Remove plugin: recource log #4372
+        - Remove particle merging  plugins #4388
```
Comment (PrometheusPi) [inline CHANGELOG.md]:
```suggestion
        - Remove particle merging plugins #4388
```

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,6 +1,354 @@
...
+    - Switch to std::void_t #4195
+    - Change user interface of free temperature functor to match free density functor #4273
+    - Change particle attribute for thermal boundaries to openPMD-conformant name #4278
+    - PIConGPU use RPATH by default #4312
+    - Remove FieldToParticleInterpolationNative #4369
+    - Remove unused file `LinearInterpolationWithUpper` #4370
+    - Switch to CI container version 3 #4378
+    - Plugin: phase space remove cuSTL #4404
+    - Clean Bunch example #4419
+    - Example foilLCT: use incident filed laser #4488
```
Comment (PrometheusPi) [inline CHANGELOG.md]:
```suggestion
    - Example foilLCT: use incident field laser #4488
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,6 +1,354 @@
...
+- Add Thermal electron benchmark #4099
+- CUDA: fix false warning #4113
+- Bugfix JUWELS Booster profile #4151
+- Add new benchmark TWEAC-FOM #4167
+- Update to cupla with alpaka 1.0.0-dev and mallocMC #4166
+- Extend TWEAC-FOM benchmark with incident field option for laser generation #4197
+- Add componentwise incident field calculation option to TWEAC-FOM benchmark #4200
+- Update cuda_memtest #4214
+- Update mallocMC to fix possible out of memory access #4220
+- CI: disable all optional dependencies for the Empty test case  #4254
```
Comment (PrometheusPi) [inline CHANGELOG.md]:
```suggestion
- CI: disable all optional dependencies for the Empty test case #4254
```

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,6 +1,354 @@
...
+- Add compilation of particle merger plugins to compile-time tests #4297
+- Update to latest cupla dev branch #4309
+- CI: add CUDA 11.3 - 11.4, HIP 4.5, 5.0 - 5.2 #4315
+- Add new openPMD-api module for hemera kepler #4320
+- Fix HIP CI tests #4353
+- Remove PIConGPU install description using spack #4375
+- KHI memory species buffer scaling based on simulation volume #4422
+- Remove warm copper example #4433
+- More variants: test KHI growth rate #4452
+- Example bunch: use incident filed laser #4449
```
Comment (PrometheusPi) [inline CHANGELOG.md]:
```suggestion
- Example bunch: use incident field laser #4449
```

**Review by PrometheusPi — CHANGES_REQUESTED** [review]
some typos, some minor readability improvements
feel free to ignore

**Comment by BrianMarre** [comment]
@psychocoderHPC @PrometheusPi CI got stuck somehow and needs a restart

File: CHANGELOG.md (hunk 2 above)
Comment (psychocoderHPC) [reply CHANGELOG.md]: It's taken from the PR headline

File: CHANGELOG.md (hunk 5 above)
Comment (psychocoderHPC) [reply CHANGELOG.md]: changed it to `Crusher hipcc` because `hipcc` should be small letters.

File: CHANGELOG.md (hunk 9 above)
Comment (psychocoderHPC) [reply CHANGELOG.md]: I removed this line because we deleted cuSTL anyway.

File: CHANGELOG.md (hunk 3 above)
Comment (PrometheusPi) [reply CHANGELOG.md]: this is still unchanged but fine with me

**Review by PrometheusPi — APPROVED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (psychocoderHPC) [reply CHANGELOG.md]: fix it was two times in the changlog.

**Pull request merged**